packages/block-editor/src/components/block-list/block.js

```diff
@@ -544,13 +559,38 @@ function BlockListBlock( {
 						ref={ breadcrumb }
 					/>
 				) }
-				{ ( shouldShowContextualToolbar || isForcingContextualToolbar.current ) && (
+
+				{ ( isCapturingDescendantToolbars || captureChildToolbar ) && (
+					// A slot made available on all ancestors of the selected Block
+					// to allow child Blocks to render their toolbars into the DOM
+					// of the appropriate parent.
+					<ChildToolbarSlot />
+				) }
+
+				{ ( ( ! ( captureChildToolbar || hasAncestorCapturingToolbars ) && shouldShowContextualToolbar ) || isForcingContextualToolbar.current ) && (
```

Comment:
I think we need to improve this logic when I type something on the navigation block so the tollbar disappears and then I press option + fn + f10, I see the toolbar render on the menu item and then move the parent. I guess forcing may also need to isForcingContextualToolbar inside the fill.

Reply:
I've updated this so that the logic for both parent and child accounts for the `isForcingContextualToolbar`. Seems to work under testing for me now.